cpp/src/jni/orc/concurrent_map.h

```diff
@@ -73,7 +69,7 @@ class concurrentMap {
 
  private:
   // starting value of the module_id.
-  static constexpr int kInitModuleId = 4;
+  static constexpr int init_module_id_ = 4;
```

Comment:
why the change back, the kInitiModuleId is what should be used for constants (also static shouldn't be required.

Reply:
non-static data member cannot be constexpr
and here is just changing the name to make it comply with Arrow C++ naming convention.

Reply:
Which naming convention are you referring to?  As as I know constants are generally of the form kInitModuleId (non-static members follow the convention you have here).

Reply:
I am a little confused about the naming convention in ARROW c++. In type.h I can find a lot static constexpr members like type_id so I thought this maybe a more consistent convention?

Reply:
Yeah, I think at times were weren't consistent with our own conventions.

Theoretically we follow the [google style guide](https://google.github.io/styleguide/cppguide.html#Constant_Names) with only a [few exceptions](https://arrow.apache.org/docs/developers/cpp.html#code-style-linting-and-ci)

Reply:
At least for now let's follow the way like init_module_id_ to be consistent with rest of the codebase.

Reply:
I might have confused the point.  Old code has both styles,  New code should (and mostly does) follow the kInitModuleId style.  At this point it would be counter-productive to change it back (I'd like to merge once CI passes) but in the future please use what is proscribed in the style guide.

Reply:
Sounds good! Thank you!